### PR TITLE
Log to the Minecraft log as all supported Log4J messages & add check functions for each log type

### PR DESCRIPTION
**Pull request by TheBunnyMan123**

### CODE REVIEW

**Comment by TheBunnyMan123** [comment]
This is also ready for review. Please ignore the minor typo I made in the commit message

File: common/src/main/java/com/thekillerbunny/goofyplugin/lua/GoofyAPI.java [hunk 1]
```diff
@@ -68,6 +72,126 @@ public String regexSub(@LuaNotNil String str, @LuaNotNil String pattern, @LuaNot
...
+    @LuaMethodDoc(
+        overloads = {
+            @LuaMethodOverload(
+                argumentTypes = {String.class},
+                argumentNames = {"data"}
+            )
+        },
+        value = "goofy.trace_to_log"
+    )
+    public void traceToLog(@LuaNotNil String data) {
```
Comment (PoolloverNathan) [inline common/src/main/java/com/thekillerbunny/goofyplugin/lua/GoofyAPI.java:149]: tracing seems like it wouldn't be useful, as it outputs the *Java* trace

File: common/src/main/java/com/thekillerbunny/goofyplugin/GoofyPlugin.java [hunk 2]
```diff
@@ -22,7 +22,7 @@
 @FiguraAPIPlugin
 @LuaWhitelist
 public class GoofyPlugin implements FiguraAPI {
-    public static final String PLUGIN_ID = "goofyfiguraplugin";
+    public static final String PLUGIN_ID = "GoofyPlugin";
     public static final Logger LOGGER = LoggerFactory.getLogger(PLUGIN_ID);
```
Comment (PoolloverNathan) [inline common/src/main/java/com/thekillerbunny/goofyplugin/GoofyPlugin.java:26]: don't capitalize IDs
```suggestion
    public static final String PLUGIN_ID = "goofyplugin";
    public static final Logger LOGGER = LoggerFactory.getLogger("GoofyPlugin");
```

**Review by PoolloverNathan — CHANGES_REQUESTED** [review]

File: common/src/main/java/com/thekillerbunny/goofyplugin/lua/GoofyAPI.java (hunk 1 above)
Comment (TheBunnyMan123) [reply common/src/main/java/com/thekillerbunny/goofyplugin/lua/GoofyAPI.java:149]: I added it to be complete with all supported options

Comment (TheBunnyMan123) [reply common/src/main/java/com/thekillerbunny/goofyplugin/lua/GoofyAPI.java:149]: So I personally want to leave it in. Didn't merge yet I wanna change one thing

Comment (PoolloverNathan) [reply common/src/main/java/com/thekillerbunny/goofyplugin/lua/GoofyAPI.java:149]: Alright makes sense

**Comment by PoolloverNathan** [comment]
whoops updating the PR destroyed the commit times; this probably won't mean anything though

**Review by PoolloverNathan — APPROVED** [review]
looks sane

**Pull request merged**